### PR TITLE
Bug 1749478: do not prematurely stop sync loop in SATokenSinger controller

### PR DESCRIPTION
**Pull request by soltysh**

### CODE REVIEW

File: test/e2e/satokensigner_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	kubeClient, err := clientcorev1.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	test.WaitForKubeControllerManagerClusterOperatorAvailableNotProgressingNotDegraded(t, configClient)
+
+	err = kubeClient.Secrets("openshift-kube-controller-manager").Delete("service-account-private-key", &metav1.DeleteOptions{})
+	require.NoError(t, err)
+
```
Comment (deads2k) [inline test/e2e/satokensigner_test.go]: do something here to ensure you go progressing before starting your wait.

**Comment by deads2k** [comment]
one comment in test.  lgtm otherwise.

File: test/e2e/satokensigner_test.go (hunk 1 above)
Comment (soltysh) [reply test/e2e/satokensigner_test.go]: Yeah, I realized I didn't even check if the secret gets recreated.

**Comment by soltysh** [comment]
@deads2k comments addressed ptal

**Pull request merged**